File: Srija/week 4/leetcode_167.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0;
        int right = numbers.size()-1;
        vector<int> ans;
        int s;
        while(left<right){
            s = numbers[left]+numbers[right];
            if(s == target){
                ans.push_back(left+1);
                ans.push_back(right+1);
                break;
            }
            else if(s < target)
                left+=1;
            else
                right-=1;
        }
        return ans;
    }
};

File: Neeraj/06-10-2022/leetcode_167.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l,r = 0,len(nums)-1
        while l<r:
            if nums[l]+nums[r] == target:
                return (l+1,r+1)
            elif nums[l]+nums[r]>target:
                r-=1
            else:
                l+=1
